### ISSUE TITLE
[5.2] Resource routes collision with 'prefix' and 'as'

### ISSUE THREAD

**Issue by EliasZ** [laravel/framework#11657]
Considering the following defined routes:

    Route::group(['namespace' => 'Group', 'prefix' => 'group', 'as' => 'group.'], function() {
        Route::get('/', function() { return 'group.index'; })->name('index');
        Route::resource('test', 'TestController');
    });

The routes list looks like this (names):

    group.index
    group.group.test.store
    group.group.test.index
    [-- other resource routes --]

Removing the `as` from the route group solves the naming of the resource routes, but now the index group has to be manually named to `group.index`:

    index
    group.test.store
    group.test.index
    [-- other resource routes --]

This is because for resource routes, the `prefix` is considered for both the URI and the name of the route.

**Comment by EliasZ** [comment]
I have created a failing unit test which - if this issue is correct - outlines the desired behavior. Should I PR this? I am capable of solving this issue but it would require some time to dive into the internals and I am certain people with a suitable knowledge about the `illuminate/routing` component wander around here.

https://github.com/EliasZ/framework/commit/7b186b90a7abae71918465ffe41c5af143695894

**Comment by EliasZ** [comment]
I fixed the bug and no other unit tests are failing. I do wonder why the prefix was added to the resource route name in the first place? Is this a breaking change or an actual bugfix? As per the docs and "regular" routes, 'as' and 'prefix' work as expected. Just not resource routes. But if people relied on this working as it does prior to this "fix", things will break for them.

ping @GrahamCampbell @taylorotwell
